src/osd/PG.cc

```diff
@@ -2970,6 +2971,14 @@ void PG::trim_peers()
 	  pg_trim_to),
 	get_osdmap()->get_epoch());
     }
+
+    // trim primary as well
+    ObjectStore::Transaction t;
+    pg_log.trim(pg_trim_to, info);
+    dirty_info = true;
+    write_if_dirty(t);
+    int tr = osd->store->queue_transaction(osr.get(), std::move(t), NULL);
+    assert(tr == 0);
```

Comment:
write_if_dirty and queueing the transaction should be handled by the caller.  The Recovered handler is in the PG state machine and is called by the peering_wq which will do that anyway.